src/fmpy/fmi3.py

```diff
@@ -697,14 +697,29 @@ def getString(self, vr, nValues=None):
         self.fmi3GetString(self.component, vr, len(vr), value, nValues)
         return list(map(lambda b: b.decode('utf-8'), value))
 
-    def getBinary(self, vr, nValues=None):
+    def getBinary(self, vr: Iterable[int], nValues: int = None) -> Iterable[bytes]:
+
         if nValues is None:
             nValues = len(vr)
+
         vr = (fmi3ValueReference * len(vr))(*vr)
+
         value = (fmi3Binary * nValues)()
+
         size = (c_size_t * len(vr))()
```

Comment:
Should be `size = (c_size_t * nValues)()` to cope with Binary variables that are arrays.